packages/gatsby/src/db/loki/nodes-query.js

```diff
@@ -152,6 +158,24 @@ const toDottedFields = (filter, acc = {}, path = []) => {
   return acc
 }
 
+// Like above, but doesn't handle $elemMatch
+const objectToDottedField = (obj, path = []) => {
```

Comment:
Looks like this is the same in `redux/run-sift` and `loki/nodes-query`? can we move it do `src/db/common/`?